src/cli/java/org/commcare/util/screen/MultiSelectEntityScreen.java

```diff
@@ -63,6 +65,45 @@ public void updateSelection(String input, TreeReference[] selectedRefs)
         }
     }
 
+    @Override
+    protected boolean shouldAutoSelect() {
+        return mNeededDatum.isAutoSelectEnabled();
+    }
+
+    @Override
+    protected void autoSelectEntities() {
+        int selectionSize = references.size();
+        if (selectionSize == 0) {
+            throw new RuntimeException(getNoEntitiesError());
+        } else if (selectionSize > maxSelectValue) {
+            throw new RuntimeException(getMaxSelectError(selectionSize));
```

Comment:
It would be nice to use custom exceptions instead of generic runtime errors.

Reply:
I see that we throw similar errors in `EntityScreen` as `CommCareSessionException` , how about utilising the same here ?

Reply:
We can though I think `CommCareSessionException` is overused.

Reply:
true, did a specific one [here](https://github.com/dimagi/commcare-core/pull/1152/commits/b3257893b491c51d85c2b70c01023d5fd2aebe79)